### PR TITLE
Improve export of DBSQL objects

### PR DESCRIPTION
**Pull request by alexott**
This includes:

* Fixing problems with listing of objects
* Correct export of queries with date/time range parameters
* Introduced separate categories for queries, dashboards & SQL endpoints to allow granular export

### CODE REVIEW

File: sql/api/query.go [hunk 1]
```diff
@@ -398,17 +400,45 @@ func (p *QueryParameterDateTimeSec) UnmarshalJSON(b []byte) error {
...
+		p.Value = p.StringValue
+	}
+}
+
+func (p *QueryParameterRangeBase) decodeQueryParameter() {
+	if v, ok := p.Value.(map[string]any); ok {
+		p.RangeValue = &DateTimeRange{Start: v["start"].(string), End: v["end"].(string)}
+	} else {
+		p.StringValue = fmt.Sprintf("%v", p.Value)
+	}
```
Comment (pietern) [inline sql/api/query.go:429]: You could set value to nil from here instead of right after this call.

**Review by pietern — COMMENTED** [review]
LGTM on the changes under `sql/`.

While listing will be much much much easier with the SDK, the type of polymorphism you need to do to make these query parameters work unfortunately remains.

File: sql/api/query.go (hunk 1 above)
Comment (alexott) [reply sql/api/query.go:429]: oh, yes - it was an artifact where I did first it as a separate function, not function on the type

**Review by nfx — APPROVED** [review]
LGTM

**Comment by alexott** [comment]
I need to run the full scale test again, and then mark it as non-draft

**Comment by alexott** [comment]
@nfx @pietern Ok, it survives export from DF :-)

**Comment by nfx** [comment]
@alexott could you rebase?

**Comment by alexott** [comment]
@nfx rebased

**Pull request merged**